test/transport/syncTrue.test.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const pino = require('../..')
+const { join } = require('node:path')
+const { readFileSync } = require('node:fs')
+const { test } = require('tap')
+const { file } = require('../helper')
+
+test('thread-stream sync true should log synchronously', async (t) => {
+  const outputPath = file()
+
+  function getOutputLogLines () {
+    return (readFileSync(outputPath)).toString().trim().split('\n').map(JSON.parse)
+  }
+
+  const transport = pino.transport({
+    target: join(__dirname, '..', 'fixtures', 'to-file-transport.js'),
+    options: { destination: outputPath },
+    sync: true
+  })
+  const instance = pino(transport)
+
+  var value = { message: 'sync' }
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
```

Comment:
I think you could do `instance.flushSync()` and then just try reading the file. It should work.

Reply:
Wouldn't calling `flushSync` force synchronously flush all messages to the stream ? That would pass even if `sync` is `false` ? I tried to simulate a event loop block immediately after calling `instance.info`. I think I need to update the loop duration by some bit. Am i missing something?

Reply:
ah no prob. This is ok

Reply:
Updated PR. Should pass now :)

Reply:
@mcollina can you retrigger the CI actions please